File: 240322/숨은 단어 찾기 2/find-hidden-words-2.py
import sys

N, M = map(int,sys.stdin.readline().split())
board = [sys.stdin.readline().strip() for _ in range(N)]

dn = [0,0]
du = [1,2]
dl = [-1,-2]

def check(row,col,dr,dc):
    temp = 0
    for i in range(2):
        nr, nc = row+dr[i], col+dc[i]
        if 0 <= nr < N and 0 <= nc < M:
            if board[nr][nc] == 'E':
                temp += 1
    return True if temp == 2 else False

count = 0
for i in range(N):
    for j in range(M):
        if board[i][j] == 'L':
            if check(i,j,dl,dn): count += 1
            if check(i,j,dl,du): count += 1
            if check(i,j,dn,du): count += 1
            if check(i,j,du,du): count += 1
            if check(i,j,du,dn): count += 1
            if check(i,j,du,dl): count += 1
            if check(i,j,dn,dl): count += 1
            if check(i,j,dl,dl): count += 1

print(count)

File: 240404/숨은 단어 찾기 2/find-hidden-words-2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

static int dr[8] = {-1,-1,0,1,1,1,0,-1};
static int dc[8] = {0,1,1,1,0,-1,-1,-1};

int main() {
    int N, M, count = 0, checker, nr, nc;
    char board[50][50];

    scanf("%d %d",&N,&M);

    for(int i = 0; i < N; i++)
        scanf("%s",&board[i]);

    for(int row = 0; row < N; row++) {
        for(int col = 0; col < M; col++) {
            if(board[row][col] == 'L') {
                for(int d = 0; d < 8; d++) {
                    checker = 0;
                    nr = row + dr[d];
                    nc = col + dc[d];
                    if(nr >= 0 && nr < N && nc >= 0 && nc < M)
                        if(board[nr][nc] == 'E') checker++;
                    nr += dr[d];
                    nc += dc[d];
                    if(nr >= 0 && nr < N && nc >= 0 && nc < M)
                        if(board[nr][nc] == 'E') checker++;
                    if(checker == 2) count++;
                }
            }
        }
    }

    printf("%d\n",count);
    return 0;
}